extension/json/json_functions/json_structure.cpp

```diff
@@ -304,12 +303,27 @@ bool TryParse(Vector &string_vector, StrpTimeFormat &format, const idx_t count)
 }
 
 bool JSONStructureNode::EliminateCandidateFormats(const idx_t vec_count, Vector &string_vector,
-                                                  const Vector &result_vector, vector<StrpTimeFormat> &formats) {
+                                                  const Vector &result_vector, DateFormatMap &date_format_map) {
 	D_ASSERT(descriptions.size() == 1 && descriptions[0].type == LogicalTypeId::VARCHAR);
+
 	const auto type = result_vector.GetType().id();
-	for (idx_t i = formats.size(); i != 0; i--) {
-		const idx_t actual_index = i - 1;
-		auto &format = formats[actual_index];
+	auto &formats = date_format_map.GetCandidateFormats(type);
+	idx_t i;
+	{
+		lock_guard<mutex> guard(date_format_map.lock);
+		i = formats.size();
+	}
+
+	for (; i != 0; i--) {
+		StrpTimeFormat format;
+		{
+			lock_guard<mutex> guard(date_format_map.lock);
```

Comment:
Instead of making the lock public can we perhaps move the code that requires the lock inside the `date_format_map`?